rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,257 @@
+[`@semantic-release/npm`](https://github.com/semantic-release/npm) it will maintain a changelog that is committed to git and published to [NPM](https://www.npmjs.com/). We will use this to help
+communicate the features that each release brings. The changelog will include the `TYPE`, `SCOPE` and `DESCRIPTION`, it
+will not include the `BODY` but it will include a link to the relevant pull request.
+
+## Maintenance Releases
+The beta version of [`semantic-release`](https://github.com/semantic-release/semantic-release) includes extensive support for publishing a maintenance
+release. There is a [detailed walk through](https://github.com/semantic-release/semantic-release/blob/916c2685c57f3490fb1e50afbf72ea8dce11e188/docs/recipes/maintenance-releases.md)
+outlining many scenarios.
+
+This functionality is not available in the latest stable release, however the beta has been evolving for in excess of a year.
+
+
+# Drawbacks
+- There is a learning curve to this new process
+- Version numbers will increase substantially if we don't plan our interfaces carefully to prevent breaking changes. I
+don't consider this to be a drawback, but we need to be aware about breaking changes. Previously we have bundled many
+breaking changes into one release.
+- Changelog format will change
+
+# Alternatives
+There are alternatives to [`semantic-release`](https://github.com/semantic-release/semantic-release) for automating the
+release process such as [np](https://github.com/sindresorhus/np)
+
+However, I could find no other solution that parses commits to work out the version number. This means that someone has
+to be involved in the release process to determine the version number. This is not sustainable and prone to errors.
+
+# Adoption strategy
+
+This RFC PR includes all changes necessary, if this RFC is accepted when it is merged it will immediately start
+releasing from the following branches `['+([0-9])?(.{+([0-9]),x}).x', 'master', 'next', 'next-major', {name: 'beta', prerelease: true}, {name: 'alpha', prerelease: true}]`  
+[[Source]](https://github.com/semantic-release/semantic-release/blob/916c2685c57f3490fb1e50afbf72ea8dce11e188/docs/usage/configuration.md#branches)
+
+Adding [`probot/semantic-pull-requests`](https://probot.github.io/apps/semantic-pull-requests/) is a manual step and
+will have to be performed by a workspace administrator.
+
+# How we teach this
+
+The proposed tooling is well established in the JavaScript community. For carbon developers
+ can use `npm run commit` to launch [`commitizen`](https://github.com/commitizen/cz-cli) which will guide them through the process.
+![](./commitizen.svg)
+
+| Type           | Description                                                                                                 | Release Type |
```

Comment:
Should we include a link to the source of these types? For example:

* https://github.com/angular/angular/blob/8.2.14/CONTRIBUTING.md#type

Reply:
These types were actually extracted from https://github.com/commitizen/conventional-commit-types as it deviates from the angular convention. In particular, it allows `chore` which angular no longer allows. It also allows the default `git revert snytax`.

Reply:
Fixed in e5d73c8bb672e0cf16b5a8ea8b7ba04b269dcb8b